### DISCUSSION TITLE
Css compatible issues - Global CSS cannot be imported from within node_modules. - React next.js with react-date-picker

### DISCUSSION THREAD

**Discussion by virkse** [vercel/next.js#19166] [Help] [answered]
# Bug report

## Bug description
If a react package writer put its css in there own component like "react-date-picker" writer put two files 
require("react-calendar/dist/Calendar.css");
require("./DatePicker.css");
in his package https://www.npmjs.com/package/react-date-picker
Now these two css, is imported in his own package, these are not related to the import in _app.js or anywere else in next.js so it gives me error as following.

./node_modules/react-calendar/dist/Calendar.css
Global CSS cannot be imported from within node_modules.
Read more: https://err.sh/next.js/css-npm
Location: node_modules/react-date-picker/dist/entry.js

## To Reproduce
npm install react-date-picker
import DatePicker from 'react-date-picker'

Steps to reproduce the behavior, please provide code snippets or a repository:

1.'npm install react-date-picker
2. import DatePicker from 'react-date-picker'
3. npm run dev

**Comment by jamesmosier** [comment, marked as answer]
Next recommends that all third party CSS should be imported separately, not within the module itself. If a module imports the CSS itself then Next had no way of deduplicating any potential duplicate component imports. 

The recommendation would be for the module publisher to publish the CSS as a separate file that can be imported alongside the imported React component.

**Comment by m-torin** [comment]
This is a known defect that isn't a high priority for fixing. "Recommended way" is good advice, but that's all it is. People will tell you that you should be doing it X way ignoring that we don't always have a choice in how something is created - it's not helpful.

Related: https://github.com/vercel/next.js/issues/12079

**Comment by tienwei** [comment]
Any workarounds on this issue if we can't really do much with the third party modules? Thanks.

**Comment by prankris** [comment]
1. Download Calendar.css and DatePicker.css from respective dist folders.
2. Copy them to public folder
3. Add following to you _app.js
   import '../public/DatePicker.css'
   import '../public/Calendar.css'
4. Use no style DatePicker
    import DatePicker from 'react-date-picker/dist/entry.nostyle';

**Reply by tienwei** [reply]
Thanks @prankris, but for a third party package which imports some css in their js files(they integrate the datepicker), in this case, the above solution won't work, right?

**Reply by hctru68** [reply]
This is the same my idea.
It's working fine for this guide.
